lib/opencensus/span.ex

```diff
@@ -0,0 +1,37 @@
+defmodule Opencensus.Span do
+  @moduledoc """
+  Elixir convenience translation of `:opencensus.span`.
+
+  Most likely to be of use while writing unit tests, or packages that deal with spans.
+  Less likely to be of use while writing application code.
+  """
+
+  alias Opencensus.SpanContext
+
+  require Record
+  @fields Record.extract(:span, from_lib: "opencensus/include/opencensus.hrl")
+  Record.defrecordp(:span, @fields)
+
+  defstruct Keyword.keys(@fields)
+
+  @doc "Get a span struct given a record."
+  @spec from(:opencensus.span()) :: %__MODULE__{}
```

Comment:
I would define `t/0` for `%__MODULE__{}` instead of using raw type, that would also help other people when implementing span-related functions.

Reply:
I'm concerned about drift from the underlying typings, especially if they fix the span attributes. Reckon it's worth the effort?

Reply:
You do not need to specify all fields, just instead of `%__MODULE__{}` define `@type t :: %__MODULE__{}` and use that. This should be enough.